### ISSUE TITLE
Performance time out error

### ISSUE THREAD

**Issue by cole007** [pixelandtonic/AssetSubfolderAccess#1]
Having unusual error with this plugin whereas the Assets page/tab of the admin area is timing out for users. Admin accounts can access this just fine but other users are timing out with the following error:

```
Fatal error: Maximum execution time of 30 seconds exceeded in /Applications/MAMP/htdocs/prior/deploy/craft/app/framework/YiiBase.php on line 128
```

**Comment by cole007** [comment]
Some digging later this does seem to be related to the `modifyAssetSources` method.

Not acquainted with the Craft API yet but should this method be returning something on completion?
Failing that should the $sources variable be updated on the fly (during the source loop) or would this be better server at the end?

**Comment by angrybrad** [comment]
Merged the performance fix into 1.1.1
